woo-razorpay.php

```diff
@@ -1703,6 +1719,79 @@ public function updateOrderAddress($razorpayData, $order)
             }
         }
 
+        //Create new user account 
+        public function newUserAccount($razorpayData, $order)
+        {
+            global $woocommerce;
+
+            if (!email_exists($razorpayData['customer_details']['email']) && isMandatoryAccCreationEnabled()) {
+
+                $name = $shpping['name'];
+                $contact = $razorpayData['customer_details']['contact'];
+
+                $email = $razorpayData['customer_details']['email'];
+                $random_password = wp_generate_password(8, false);
+
+                //create user name with the help default woocommerce function
+                $username = wc_create_new_customer_username( $email , $name);
+                $userId  = wp_create_user( $username, $random_password, $email );
+                $user = get_user_by('id', $userId);
+
+                update_post_meta($order->id, '_customer_user', $userId);
+
+                // Get all WooCommerce emails Objects from WC_Emails Object instance
+                $emails = wc()->mailer()->emails;
+
+                // Send WooCommerce "Customer New Account" email notification with the password
+                $emails['WC_Email_Customer_New_Account']->trigger( $userId, $random_password, true );
+
+                update_user_meta( $userId, 'shipping_email', $email);
+                update_user_meta( $userId, 'shipping_phone', $contact );
+
+                if (isset($razorpayData['customer_details']['shipping_address']))
```

Comment:
@poojarazor Can we reuse [this](https://github.com/razorpay/razorpay-woocommerce/blob/73ccb75fd44e66a5e33fbc7b9268abf5cba93342/woo-razorpay.php#L1663) to remove the code duplication!

Reply:
Tried but it's not updating address every time when we use "UpdateOrderAddress" function , might be assigning userId taking time